app/src/main/java/com/benavalli/arcmovies/features/adapter/MoviesAdapter.kt

```diff
@@ -0,0 +1,44 @@
+package com.benavalli.arcmovies.features.adapter
+
+import android.content.Context
+import android.databinding.DataBindingUtil
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import com.benavalli.arcmovies.BuildConfig
+import com.benavalli.arcmovies.R
+import com.benavalli.arcmovies.api.config.BuildConstants
+import com.benavalli.arcmovies.databinding.ListMoviesBinding
+import com.benavalli.arcmovies.domain.UpcomingMovies.UpcomingMovie
+import com.squareup.picasso.Picasso
+
+class MoviesAdapter : ArrayAdapter<UpcomingMovie> {
+
+    private var movies = ArrayList<UpcomingMovie>()
+    private var adapterContext: Context? = null
+
+    constructor(context: Context, movies: ArrayList<UpcomingMovie>): super(context, 0, movies) {
+        this.movies = movies
+        this.adapterContext = context
+    }
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
+        val inflater = LayoutInflater.from(context)
+        val binding : ListMoviesBinding = DataBindingUtil
+                .inflate(inflater, R.layout.list_movies, null, false)
+
+        binding.listMovieTitle.text = movies[position].name
+        binding.listMovieGenresValue.text = movies[position].genre.map{ it }.joinToString(", ")
```

Comment:
This gender issue is a little trick. Do you think is the right way display the id number? Take a look to genre list endpoint

Reply:
Yes, I saw during the development, but i left to do it later by the time,  i will call the genre endpoint and join the result with the upcoming movies response in the repository.

Reply:
fixed